src/popup/actions/popup-action-message-creator.ts

```diff
@@ -57,19 +56,19 @@ export class PopupActionMessageCreator {
         });
     }
 
-    public openDetailsView(
+    public async openDetailsView(
```

Comment:
Do you have an understanding of why this message creator needs to be async but it's (presumably) safe to leave the other ones as sync? It feels like dispatchMessage being sync might end up causing a lot of hard to diagnose bugs if we leave it as is.

Reply:
I'm not sure if leaving `dispatchMessage` as is would cause a lot of issues: dispatchMessage is where we cross boundaries between popup/details-view/etc. to background so the async work is really just making sure the message is sent and after that, nothing else is guaranteed (the rest of which is async as well).

In this particular case, the reason we run into an issue is that even just the dispatch of the message is not completed in the action-message-creator before we kill the popup, resulting in it never being completed.

Reply:
Makes sense. Since there are so many references to dispatchMessage I assume it might not be a good use of time to audit all of them to check that there isn't another situation like this... but might be good to keep in mind going forward.